src/tree/localProject/ConfigGroupTreeItem.ts

```diff
@@ -3,37 +3,83 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
+import { pathExists, readdir } from "fs-extra";
+import { basename, join } from "path";
+import { ThemeIcon } from "vscode";
 import { AzExtParentTreeItem, AzExtTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
 import { localize } from "../../utils/localize";
-import { treeUtils } from "../../utils/treeUtils";
-import { ConfigTreeItem } from "./ConfigTreeItem";
-import { LocalProjectTreeItem } from "./LocalProjectTreeItem";
+import { parseYamlFile } from "../../utils/yamlUtils";
+import { EnvironmentTreeItem } from "../EnvironmentTreeItem";
+import { GitHubConfigTreeItem } from "./ConfigTreeItem";
 
-export class ConfigGroupTreeItem extends AzExtParentTreeItem {
-    public static contextValue: string = 'azureStaticConfigGroup';
-    public contextValue: string = ConfigGroupTreeItem.contextValue;
-    public readonly label: string = localize('config', 'Configuration');
-    public parent: LocalProjectTreeItem;
+export type BuildConfig = 'app_location' | 'api_location' | 'output_location' | 'app_artifact_location';
 
-    public constructor(parent: LocalProjectTreeItem) {
+export type BuildConfigs = {
+    'app_location': string,
+    'api_location': string,
+    'output_location': string | undefined,
+    'app_artifact_location': string | undefined
+}
+
+export class GitHubConfigGroupTreeItem extends AzExtParentTreeItem {
+    public static contextValue: string = 'azureStaticGitHubConfigGroup';
+    public contextValue: string = GitHubConfigGroupTreeItem.contextValue;
+    public readonly label: string = localize('gitHubConfig', 'GitHub Configuration');
+    public parent: EnvironmentTreeItem;
+    public yamlFilePath: string;
+    public buildConfigs: BuildConfigs | undefined;
+
+    public constructor(parent: EnvironmentTreeItem, yamlFilePath: string, buildConfigs: BuildConfigs) {
         super(parent);
+        this.yamlFilePath = yamlFilePath;
+        this.buildConfigs = buildConfigs;
+        this.id = `${parent.id}-${this.yamlFilePath}`;
+    }
+
+    public static async createGitHubConfigGroupTreeItems(parent: EnvironmentTreeItem): Promise<GitHubConfigGroupTreeItem[]> {
+        if (parent.localProjectPath && parent.inWorkspace) {
+            const treeItems: GitHubConfigGroupTreeItem[] = [];
```

Comment:
nit: Define treeItems outside of the if conditional, and just return treeItems outside of the if statement (on line 56)